Zend/tests/enum/instanceof-scalarenum.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+Auto implement ScalarEnum interface
```

Comment:
I wonder if it'd be useful to forbid the use of ScalarEnum/UnitEnum by classes/interfaces that aren't enum

(e.g. so that a static analyzer could warn about `function example(UnitEnum $x) { $x->missingProp = 'value'; }` with more certainty, or so that opcache could more aggressively optimize uses of enums in the future)

Then again, I don't think checking against those enums could be very common in practice (maybe for immutable data structures?)

> The ScalarEnum interface additionally has a property `value` that returns the scalar value for a given case.

The RFC phrasing may be clearer as "instances of classes implementing the ScalarEnum interface have..."

Reply:
Searching the code for Traversable(Iterator/IteratorAggregate only) or Throwable(Exception/Error) may or may not help in implementing that restriction

```
php > class X implements Traversable{}

Fatal error: Class X must implement interface Traversable as part of either Iterator or IteratorAggregate in Unknown on line 0
```

Reply:
> I wonder if it'd be useful to forbid the use of ScalarEnum/UnitEnum by classes/interfaces that aren't enum (or add a test)

Yes, we've discussed this before and I totally forgot to implement it :smile: